### PR TITLE
FixedIssue346

### LINKED ISSUE #346: Decouple bucket name from project id in fileStore
### The problem faced currently?
There is no way to decouple the bucket name of `amazon-s3` from the project id
### How can we solve it?
We can solve this by providing a bucket name in the config file or either in the request directly. If we are mentioning the bucket name in the request itself, then there needs to be some form of mechanism to mention which bucket names are allowed
>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
@YourTechBud Done

Fixes #346

### CODE REVIEW

File: modules/crud/sql/read.go [hunk 1]
```diff
@@ -69,18 +69,30 @@ func (s *SQL) Read(ctx context.Context, project, col string, req *model.ReadRequ
...
+	case utils.Count:
+		query = query.Select(goqu.COUNT("*"))
+	case utils.Distinct:
+		distinct := req.Options.Distinct
+		if distinct == nil {
+			return nil, utils.ErrInvalidParams
+		}
+		query = query.SelectDistinct(*distinct)
+	case utils.One, utils.All:
+		query = query.Select(selArray...)
```
Comment (YourTechBud) [inline modules/crud/sql/read.go]: What will happen if select array is empty? Do we get the results?

File: modules/crud/sql/read.go [hunk 2]
```diff
@@ -101,6 +113,27 @@ func (s *SQL) Read(ctx context.Context, project, col string, req *model.ReadRequ
...
+			return nil, err
+		}
+
+		switch s.GetDBType() {
+		case utils.MySQL, utils.Postgres:
+			mysqlTypeCheck(rowTypes, mapping)
+		}
+
+		for _, v := range mapping {
+			return v, nil
```
Comment (YourTechBud) [inline modules/crud/sql/read.go]: What are we exactly returning here? The count number? I think we should return an object with key count. Whats the mongo api returning?

File: modules/crud/sql/read.go (hunk 1 above)
Comment (AliabbasMerchant) [reply modules/crud/sql/read.go]: Yes

File: modules/crud/sql/read.go (hunk 2 above)
Comment (AliabbasMerchant) [reply modules/crud/sql/read.go]: Yes, we are returning a number. The mongo api also returns a count number.

**Pull request merged**